libsolidity/formal/SMTChecker.cpp

```diff
@@ -702,6 +704,20 @@ void SMTChecker::resetVariables(vector<Declaration const*> _variables)
 	}
 }
 
+void SMTChecker::mergeVariables(vector<Declaration const*> const& _variables, smt::Expression const& _condition, VariableSequenceCounters const& _countersEndTrue, VariableSequenceCounters const& _countersEndFalse)
+{
+	for (auto const* decl: _variables)
+	{
+		int trueCounter = _countersEndTrue.at(decl);
```

Comment:
I think we should skip all variables where the true counter and the false counter are equal.

Reply:
I think the only way this can happen is if the variable isn't touched in any of the branches, in which case it wouldn't be part of `_variables`. This is because even though `m_currentSequenceCounter` is reset after each branch, `m_nextFreeSequenceCounter` is not, so the index that the variable gets for an assignment is different in each branch.

Reply:
Also, I just realized that some SMT `ite` vars are being created twice since `touchedVariables` is a `vector` and variables that are touched inside both branches are inserted twice. Fixing that now.

Reply:
Ah ok, that makes sense. In that case: Could you add a `solAssert` check to see that `_variables` equals the set of sequence counters that are different?

Reply:
Added `solAssert(trueCounter != falseCounter, "");` inside the loop, should lead to the same I think.